### PR TITLE
Implementing an abstract class for genetic algorithms

### PR DESCRIPTION
**Pull request by alialrafei**
I have coded the abstract class and also built a simple genetic algorithm.
in my code i have used deap lib in my cross over function and can use any type of deap's cross over functions also adopted the Evoparms in my code to be easy for us to work with different data types.
I hope every one test the code as will as report for bugs and issues if found.

### CODE REVIEW

**Comment by alialrafei** [comment]
@samehkamaleldin  Sorry it was a mistake the develop branch is up-to-date

File: libevolve/ga/SGA.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import random
+
+from ._base import GeneticAlgorithm
+from deap import base, algorithms, creator, tools
+from libevolve.libevolve.common._base import EvoCategoricalParam,EvoIntParam,EvoFlagParam, EvoFloatParam
```
Comment (samehkamaleldin) [inline libevolve/ga/SGA.py]: you should import from `libevolve.libevolve.common`
```suggestion
from libevolve.libevolve.common import *
```

File: libevolve/GA_API.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+
+
+
+
+
+
+
+def objective_fn(param,w):
+    """ Sample objective function for evolutionary algorithm
```
Comment (samehkamaleldin) [inline libevolve/GA_API.py:18]: Use `numpy` docstring instead of the `rStructredText` for consistency.

File: libevolve/GA_API.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+
+
+
+
+
+
+
+
+test_app = Flask(__name__)
```
Comment (samehkamaleldin) [inline libevolve/GA_API.py:89]: Why do you use flask here? The flask API will be in another project. This codebase will only be for generic genetic algorithms

File: libevolve/GA_API.py (hunk 2 above)
Comment (alialrafei) [reply libevolve/GA_API.py:18]: i copied it from the code you previously sent to us

File: libevolve/GA_API.py (hunk 3 above)
Comment (alialrafei) [reply libevolve/GA_API.py:89]: i wanted to test the code only and i didnt complete the code required for flask api

Comment (samehkamaleldin) [reply libevolve/GA_API.py:89]: This code was intended to be for demonstration only. Anyway, it can be safely removed from this repository.

Comment (alialrafei) [reply libevolve/GA_API.py:89]: yes for sure

File: libevolve/GA_API.py (hunk 2 above)
Comment (samehkamaleldin) [reply libevolve/GA_API.py:18]: Ah. Anyway, try to change your editor to `numpy` docstring instead of the `rStructredText` for the future code snippets.

**Pull request closed without merging**